tests/api/printf.cpp

```diff
@@ -227,4 +227,34 @@ TEST_F(WithCommandQueue, PrintfMissingLengthModifier) {
     ASSERT_STREQ(printf_buffer, message);
 }
 
+void printf_callback(const char* buffer, size_t len, size_t complete,
+                     void* user_data) {
+    ASSERT_EQ(strlen(buffer), len);
+    printf("Received data: %s (length: %zu)\n", buffer, len);
+}
```

Comment:
This is not really testing anything. If we return an empty buffer, it would pass.
We cannot rely on visual check

Reply:
what would be an appropriate way to validate the contents of the buffer?

Reply:
Comparing it to the expectation